crates/bevy_math/src/ops.rs

```diff
@@ -442,6 +501,68 @@ mod libm_ops {
     pub fn atanh(x: f32) -> f32 {
         libm::atanhf(x)
     }
+
+    /// Calculates the least nonnegative remainder of `self (mod rhs)`.
+    ///
+    /// Precision is specified when the `libm` feature is enabled.
+    #[inline(always)]
+    pub fn rem_euclid(x: f32, y: f32) -> f32 {
+        let result = libm::remainderf(x, y);
+
+        if result < 0. {
+            result + y
+        } else {
+            result
+        }
+    }
```

Comment:
I noticed when experimenting with the `curve` feature that `libm::remainderf` was returning a negative value for `remainderf(0.99, 1.0)`. I can't find any documentation stating that it _must_ return a non-negative value, so I am going to assume it's intended behaviour. As such, we must check if it is negative and return the next value on the modulus.

Reply:
It's explained [here](https://sourceware.org/newlib/libm.html#remainder) that `remainderf` produces something in the range `-y/2 .. y/2` :)

Reply:
Thanks for pointing that out! I might change my shim here to instead always add `x/2.0` to avoid the branch.

Reply:
I tried adding `y/2.` in all cases but it did not provide results consistent with `f32::rem_euclid`, so I've opted to keep this version instead. It will only be used on platforms without the `std` feature enabled, so there is no change in behaviour for the existing Bevy targets.